### PR TITLE
Fix kind image loading in some environments

### PR DESCRIPTION
**Pull request by mangelajo**

### CODE REVIEW

File: hack/deploy_with_helm.sh [hunk 1]
```diff
@@ -53,9 +53,52 @@ HELM_SETS="${HELM_SETS} --set global.baseDomain=${BASEDOMAIN}"
...
+  else
+    echo "Saving ${image} to ${tar_filename}..."
+
+    # If the image is not local we may need to pull it first
+    if [[ "${image}" != localhost* ]]; then
+      podman pull "${image}"
+    fi
+
+    # Save to tar file
+    podman save "${image}" -o "${tar_filename}"
```
Comment (NickCao) [inline hack/deploy_with_helm.sh:84]: We don't need a temporary file here:
```bash
podman save <image> | kind load image-archive /dev/stdin
```

Comment (mangelajo) [reply hack/deploy_with_helm.sh]: Done.

File: hack/deploy_with_helm.sh [hunk 2]
```diff
@@ -53,9 +53,41 @@ HELM_SETS="${HELM_SETS} --set global.baseDomain=${BASEDOMAIN}"
...
+  local image=$1
+
+  # First, try to load the image directly
+  if kind load docker-image "${image}" --name jumpstarter; then
+    echo "Image ${image} loaded successfully."
+    return
+  fi
+
+  # If that fails, we have the workaround in place
+  echo "Saving ${image} to ${tar_filename}..."
```
Comment (NickCao) [inline hack/deploy_with_helm.sh:71]: Some leftover.

Comment (mangelajo) [reply hack/deploy_with_helm.sh:71]: Ooops, true, thanks!

Comment (mangelajo) [reply hack/deploy_with_helm.sh]: Done.

**Review by NickCao — APPROVED** [review]

**Pull request merged**